extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+class CuratedContentValidator {
+	const LABEL_MAX_LENGTH = 48;
+
+	private $errors;
+	private $titles;
+	private $hadEmptyLabel;
+
+	public function __construct( $data ) {
+		$this->errors = [ ];
+		$this->titles = [ ];
+		$this->hadEmptyLabel = false;
+
+		// validate sections
+		foreach( $data as $section ) {
+			if ( !empty( $section['featured'] ) ) {
+				$this->validateFeaturedSection( $section );
+			} else {
+				$this->validateSection( $section );
+			}
+		}
+		// also check section for duplicate title
+		foreach( array_count_values( $this->titles ) as $title => $count ) {
```

Comment:
We put space between block name and bracket. No space between function name and bracket, though.